linkerd/app/outbound/src/lib.rs

```diff
@@ -244,3 +261,85 @@ pub fn trace_labels() -> HashMap<String, String> {
     l.insert("direction".to_string(), "outbound".to_string());
     l
 }
+
+// === impl ParentRef ===
+
+impl std::ops::Deref for ParentRef {
+    type Target = policy::Meta;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<EndpointRef> for ParentRef {
+    fn from(EndpointRef(meta): EndpointRef) -> Self {
+        Self(meta)
+    }
+}
+
+// === impl RouteRef ===
+
+impl std::ops::Deref for RouteRef {
+    type Target = policy::Meta;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+// === impl BackendRef ===
+
+impl std::ops::Deref for BackendRef {
+    type Target = policy::Meta;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<ParentRef> for BackendRef {
+    fn from(ParentRef(meta): ParentRef) -> Self {
+        Self(meta)
+    }
+}
+
+impl From<EndpointRef> for BackendRef {
+    fn from(EndpointRef(meta): EndpointRef) -> Self {
+        Self(meta)
+    }
+}
+
+// === impl EndpointRef ===
+
+impl std::ops::Deref for EndpointRef {
+    type Target = policy::Meta;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl EndpointRef {
+    fn new(md: &Metadata, port: NonZeroU16) -> Self {
```

Comment:
nit, take it or leave it (definitely not important): since this always returns metadata for a Pod, should this maybe be 
```suggestion
    fn new_pod(md: &Metadata, port: NonZeroU16) -> Self {
```
or something?